### ISSUE TITLE
Driver exception during data transfer

### ISSUE THREAD

**Issue by togruen** [linux4wilc/driver#121]
Hi,

i want to use a ATWILC1000 device with an Zynq 7030 for Wifi Connectivity. The ATWILC1000 is connected via SDIO Interface.
The Zynq is running a Petalinux with neccessary changes in kernel, rootfs and devicetree according to the ATWILs Linux Porting Guide.
When the Zynq boots it shows, that the driver is loaded. With correct WPA settings, it can connect to an existing Wifi network. Everything seems okay, but when large files are copied with SCP, the connection fails.
This error occurs also, when an application is debugged via tcf from Vitis SDK.

Below is the errormessage that is printed in the console:
```
user.alert kernel: Unable to handle kernel NULL pointer dereference at virtual address 00000108
user.alert kernel: pgd = 7abc4b3a
user.alert kernel: [00000108] *pgd=00000000
user.emerg kernel: Internal error: Oops - BUG: 5 [#1] PREEMPT SMP ARM
user.warn kernel: Modules linked in: uio_pdrv_genirq
user.warn kernel: CPU: 0 PID: 404 Comm: K_TXQ_TASK Not tainted 5.10.0-xilinx-v2021.2 #1
user.warn kernel: Hardware name: Xilinx Zynq Platform
user.warn kernel: PC is at wilc_wlan_handle_txq+0x12c/0x590
user.warn kernel: LR is at _raw_spin_unlock_irqrestore+0x18/0x50
user.warn kernel: pc : [<c050ada8>]    lr : [<c07b23f8>]    psr: 20000013
user.warn kernel: sp : c20ebe00  ip : 00000000  fp : c20ebe2c
user.warn kernel: r10: 00000058  r9 : c20ebe30  r8 : 00000001
user.warn kernel: r7 : 00000100  r6 : 00000000  r5 : c18b8500  r4 : 00000000
user.warn kernel: r3 : 00000001  r2 : 00000000  r1 : 80000013  r0 : c18b8630
user.warn kernel: Flags: nzCv  IRQs on  FIQs on  Mode SVC_32  ISA ARM  Segment none
user.warn kernel: Control: 18c5387d  Table: 022e004a  DAC: 00000051
user.emerg kernel: Process K_TXQ_TASK (pid: 404, stack limit = 0x462187b9)
user.emerg kernel: Stack: (0xc20ebe00 to 0xc20ec000)
user.emerg kernel: be00: c07b244c c18bc000 c22f0000 c18b8630 c18b861c c18b8c90 c18b8630 a0000013
user.emerg kernel: be20: c20ebf60 00000000 00000000 00000016 c19a32c0 00000002 c07af134 c10dd2c0
user.emerg kernel: be40: c07b14a8 c20ebe9c c22d4044 c20ebe68 c07b2448 c07b244c 60000013 ffffffff
user.emerg kernel: be60: 00000051 00000000 00000000 c0139f3c 00000000 00000002 c0c00000 00000000
user.emerg kernel: be80: deb4e300 c10dd2c0 c0c07580 c20ea000 c07af37c c10dd660 c20ebedc c07af134
user.emerg kernel: bea0: deb5d300 c100ba18 c20ea000 c0c043b4 00000001 98e93198 deb5d300 c10dd2c0
user.emerg kernel: bec0: c20ea000 c20ea000 00000002 c20ea000 c20ea000 c07b14a8 c20ebeec c07af37c
user.emerg kernel: bee0: 7fffffff c18b867c 00001388 c07b14d4 c214b340 deb5d300 c20ea000 c214b7dc
user.emerg kernel: bf00: c214b340 c013b8f8 c0c00000 98e93198 deb4e300 7fffffff c18b867c c18b8680
user.emerg kernel: bf20: 7fffffff c07afcf0 c10dd2c0 98e93198 c20ebf2c c18b719c c18b8500 c20ea000
user.emerg kernel: bf40: 00000000 c18b868c c18b8540 c18b867c 00001388 c05060a8 00000001 c18b8524
user.emerg kernel: bf60: 00000000 98e93198 60000013 c20ea000 c22bce00 c0506050 c18b8500 c22bce24
user.emerg kernel: bf80: c22bce40 c2259cbc 00000000 c01348c0 c22bce40 c013477c 00000000 00000000
user.emerg kernel: bfa0: 00000000 00000000 00000000 c0100148 00000000 00000000 00000000 00000000
user.emerg kernel: bfc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
user.emerg kernel: bfe0: 00000000 00000000 00000000 00000000 00000013 00000000 00000000 00000000
user.emerg kernel: [<c050ada8>] (wilc_wlan_handle_txq) from [<c05060a8>] (wilc_txq_task+0x58/0x120)
user.emerg kernel: [<c05060a8>] (wilc_txq_task) from [<c01348c0>] (kthread+0x144/0x150)
user.emerg kernel: [<c01348c0>] (kthread) from [<c0100148>] (ret_from_fork+0x14/0x2c)
user.emerg kernel: Exception stack(0xc20ebfb0 to 0xc20ebff8)
user.emerg kernel: bfa0:                                     00000000 00000000 00000000 00000000
user.emerg kernel: bfc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
user.emerg kernel: bfe0: 00000000 00000000 00000000 00000000 00000013 00000000
user.emerg kernel: Code: e28db02c e1a0a006 e1a0900b e1a08006 (e5973008) 
user.warn kernel: ---[ end trace c3660c8ce3e6fa78 ]---
```

Can anyone help me how to fix this?

**Comment by ajaykathat** [comment]
Hi @togruen,

Which version of the WILC driver was used to test this scenario.

In case, it was not tested using WILCv15.6 [(driver source link)](https://github.com/linux4microchip/linux/commit/32e7037bd862187cc63369763240df5f4f167e0e) & Firmware ([link](https://github.com/linux4wilc/firmware/tree/wilc_linux_15_6)) then it's worth a try because there was a commit[(32e7037bd862187cc63369763240df5f4f167e0e)](https://github.com/linux4microchip/linux/commit/32e7037bd862187cc63369763240df5f4f167e0e) done to handle a similar exception.

If the issue is observed even with the latest driver/FW version in your setup then, please open a sales force ticket for this issue with complete logs.
http://microchipsupport.force.com

Regards,
Ajay

**Comment by togruen** [comment]
Hi @ajaykathat

thank you for your help. The issue has to do with the commit you mentioned. First we tried to patch the driver included in the Petalinux Kernel to the newest version without success.
We saw just yesterday the new Petalinux Verion 2022.1 was released including a newer version of the driver. With this build filecopy and tcf debugging work perfectly. I think if we had spend more time, the patch of the driver would have been a solution as well.

Regards,
Tobias
